router/statistics/query_time_statistics.go

```diff
@@ -25,12 +26,14 @@ type statistics struct {
 	TimeData          map[string]*startTimes
 	Quantiles         []float64
 	NeedToCollectData bool
+	lock              sync.RWMutex
```

Comment:
Maybe you actually need [sync.Map](https://pkg.go.dev/sync#Map)

Reply:
I read about it, it seems that sync.Map is optimal to use when we mostly read from map and have many cores. Is it the case?